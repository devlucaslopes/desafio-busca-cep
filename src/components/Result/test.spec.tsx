import React from 'react'
import { render, screen } from '@testing-library/react'

import { Result } from '.'

describe('<Result />', () => {
  it('should make snapshot', () => {
    const { container } = render(
      <Result
        zipcode="123456"
        state="MG"
        city="Belo Horizonte"
        address="Rua XPTO"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it('should render correctly', () => {
    render(
      <Result
        zipcode="123456"
        state="MG"
        city="Belo Horizonte"
        address="Rua XPTO"
      />
    )

    expect(screen.getByText(/123456/i)).toBeInTheDocument()
    expect(screen.getByText(/mg/i)).toBeInTheDocument()
    expect(screen.getByText(/belo horizonte/i)).toBeInTheDocument()
    expect(screen.getByText(/rua xpto/i)).toBeInTheDocument()
  })
})
