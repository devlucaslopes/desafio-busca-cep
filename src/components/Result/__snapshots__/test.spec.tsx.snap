// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should make snapshot 1`] = `
<div>
  <ul>
    <li>
      <strong>
        CEP:
      </strong>
      <span>
        123456
      </span>
    </li>
    <li>
      <strong>
        Estado:
      </strong>
      <span>
        MG
      </span>
    </li>
    <li>
      <strong>
        Cidade:
      </strong>
      <span>
        Belo Horizonte
      </span>
    </li>
    <li>
      <strong>
        Logradouro:
      </strong>
      <span>
        Rua XPTO
      </span>
    </li>
  </ul>
</div>
`;
